### PR TITLE
Foraging: final deer population features

### PR DESCRIPTION
**Pull request by JamesTev**
# Summary

Fixed deer population code so it updates as it should. Also created a way to link the probability of catching a deer to the instantaneous deer population to (so that agents can't just hunt deer into oblivion). 

## Additional Information

Also added input and output scalers for all foraging types. This will allow us to scale the amount of resources required to reach various foraging utility tiers (number of deer/fish one can hunt), as well as scaling the return resources so that they are in an appropriate range relative to other resource quantities (cost of living, salaries etc)

## Test Plan

- all tests passing

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: internal/common/foraging/deerhunt.go [hunk 1]
```diff
@@ -49,3 +57,29 @@ func deerReturn(params deerHuntParams) shared.Resources {
...
+
+// getPopulationLinkedProbability returns the Bernoulli probability of catching a deer given the current running deer population.
+// The dynamics and variables implemented in this function are documented in the README in this package.
+func (d DeerHunt) getPopulationLinkedProbability(dhConf config.DeerHuntConfig, population uint) float64 {
+	pCritical := 1.0
+	pMax := float64(dhConf.MaxDeerPopulation) / float64(dhConf.MaxDeerPerHunt)
+	thetaCritical := dhConf.ThetaCritical
+	thetaMax := dhConf.ThetaMax
+	p := float64(population) / float64(dhConf.MaxDeerPerHunt)
+	alpha := (thetaMax - thetaCritical) / (pMax - float64(pCritical))
```
Comment (NGovani) [inline internal/common/foraging/deerhunt.go:69]: No issues, just wanted to say I'm very impressed with this. I don't know how you came up with it all.

**Review by NGovani — APPROVED** [review]
lgtm

File: internal/common/baseclient/forage.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ func (c *BaseClient) DecideForage() (shared.ForageDecision, error) {
 	ft := int(math.Round(rand.Float64())) // 0 or 1 with equal prob.
 	return shared.ForageDecision{
 		Type:         shared.ForageType(ft),
-		Contribution: shared.Resources(rand.Float64() * 5),
+		Contribution: shared.Resources(rand.Float64() * 20),
```
Comment (lhl2617) [inline internal/common/baseclient/forage.go:16]: why not a multiple of how many resources it has? :)

**Pull request merged**